user-script.js

```diff
@@ -0,0 +1,175 @@
+// ==UserScript==
+// @name         Google Meet Leader Election
+// @namespace    http://tampermonkey.net/
+// @version      1.0
+// @description  When multiple Tabs/Windows join the same Meet, one is elected as Audio I/O Source, all others are muted. To be used together with https://github.com/seibert-media/googlemeet-presentation-css
+// @author       Peter Körner <pkoerner@seibert-media.net>
+// @match        https://meet.google.com/*
+// @icon         data:image/gif;base64,R0lGODlhAQABAAAAACH5BAEKAAEALAAAAAABAAEAAAICTAEAOw==
+// @grant        none
+// ==/UserScript==
+
+(function() {
+  const broadcastInterval = 500;
+  const purgeTimeout = broadcastInterval * 1.5;
+
+  // Logging
+  function log(s) {
+    const args = arguments;
+    args[0] = 'LeaderElection: ' + s;
+    console.log.apply(console, args);
+  }
+
+  // State
+  const contexts = {}
+
+  // Events
+  const channelName = location.host + location.pathname;
+  const contextId = Math.random().toString(36).substr(2);
+  log('channelName', channelName);
+  log('contextId', contextId);
+
+  const channel = new BroadcastChannel(channelName)
+  channel.addEventListener('message', (e) => onMessage(e.data))
+  window.addEventListener('DOMContentLoaded', () => transmitExistence());
+  window.setInterval(() => {
+    transmitExistence();
+    purgeOldContexts();
+    updateLeaderState();
+  }, 500);
+
+  window.addEventListener('beforeunload', function (e) {
+    e.preventDefault();
+    e.stopImmediatePropagation();
+    e.stopPropagation();
+    e.returnValue = 'Don\'t leave me this way...';
+  });
```

Comment:
Won't this prevent closing, but still disable sending "i'm here" messages?

Reply:
Nein, wie kommst du darauf? Wir fangen das `beforeunload` event und verhindern, dass es vom Meet oder vom Browser verarbeitet wird. Die Semantik ist hier aus historischen Gründen etwas broken – sobald was in `.returnValue` steht, bricht der Browser das schließen ab, zeigt die eigentliche Nachricht aber nicht (mehr) an (das wurde früher oft missbräuchlich benutzt).

Reply:
Ne, war nur noch nicht wach. Alles gut.